[IV-23-24] Objetivo 6

migueruiz opened the pull request:
* [x] ¿Se han establecido criterios *a priori* para elegir el sistema de
integración continua más conveniente y se han calificado los sistemas que
cumplan esos requisitos según esos criterios?
* [x] ¿Se han examinado varios sistemas de integración continua?
* [x] ¿Se han configurado varios sistemas de integración continua?
* [x] ¿Se ha configurado correctamente el *Checks API* en los sistemas en los que
   sea necesario para que aparezca correctamente en GitHub (y se pueda comprobar
   desde los tests)?
* [x] ¿Uno de los sistemas configurados permite comprobar cuales son las versiones
   dellenguaje con las que funciona correctamente nuestra aplicación?
* [x] ¿Se escogen de forma adecuada las versiones del lenguaje que se testean,
   tanto en el sistema de CI como en el contenedor Docker?
* [x] ¿Se han justificado correctamente las versiones del lenguaje que se están
testeando y se ha comprobado que no se comprueban varias veces lo mismo (la
misma versión en CI y en el contenedor Docker en otro sistema CI?

migueruiz committed:
Añado fichero config.yml para CircleCI. closes #45.

Añado toda la configuración necesaria para que funcione CircleCI en el proyecto.

migueruiz committed:
Añado doc/eleccion_imagen.md. closes #44

Añado el archivo donde comento las opciones y eleccion del servicio de CI.

migueruiz committed:
Añado fichero node.yml para GitHub Actions. closes #45.

Añado toda la configuración necesaria para que funcione GitHub Actions en el proyecto.

migueruiz committed:
Modifico el contenido de iv.yaml. closes #48

Añado el contenido necesario para el funcionamiento de Circle CI y de github Actions.

migueruiz committed:
Modifico el contenido de README.md. closes #47

Añado el enlace a la elección de los sistemas de integración continua.

migueruiz committed:
Modifico el contenido de README.md. closes #47

Añado el enlace a la elección de los sistemas de integración continua.

migueruiz committed:
Modifico el contenido de README.md. closes #47

Añado el enlace a la elección de los sistemas de integración continua.

migueruiz committed:
Añado el fichero node.yml para GitHub Actions. closes #46.

Añado toda la configuración necesaria para que funcione GitHub Actions en el proyecto.

migueruiz committed:
Modifico el fichero node.yml para GitHub Actions. closes #46.

Modifico error Typo.

migueruiz committed:
Modifico el fichero README.md closes #47

Arreglo el para que se vea mejor.

migueruiz committed:
Modifico el fichero node.yml para GitHub Actions. closes #46.

Cambio el sistema operativo, ya que he tenido una confusión pensando que era el SO anfitrion.

migueruiz committed:
Modifico el fichero node.yml para GitHub Actions. closes #46.

Cambio el sistema operativo, ya que he tenido una confusión.

danieeeld2 commented:
¿Está para revisar @migueruiz? Como sale en draft.

migueruiz commented:
Aún estoy en proceso de terminarlo, cuando termine te aviso 🫡

migueruiz committed:
Modifico el fichero /doc/eleccion_ci.md. closes #44.

Especifico la versión de node que he utilizado.

migueruiz commented:
@danieeeld2 Listo para la revisión!

danieeeld2 reviewed (approved):
:+1: En general muy bien, se podría haber intentado probar alguna herramienta más, pero es verdad que la mayoría son complicadas, lo digo por experiencia propia de ayer jajaja. No sé como será para `node`, pero en `go` la verdad que `cirrus` y `circle` eran gemelas. Como todavía no lo ha usado mucha gente no creo que tengas problemas :). Si tuvieras alguno y te pidiese otra, puedes probar `Travis`, hasta hace poco tenía buen soporte, pero ahora han cambiado de equipo y está algo peor (pj en `go` no tenían la última versión, pero creo que para `node` sí)

migueruiz commented:
> 👍 En general muy bien, se podría haber intentado probar alguna herramienta más, pero es verdad que la mayoría son complicadas, lo digo por experiencia propia de ayer jajaja. No sé como será para `node`, pero en `go` la verdad que `cirrus` y `circle` eran gemelas. Como todavía no lo ha usado mucha gente no creo que tengas problemas :). Si tuvieras alguno y te pidiese otra, puedes probar `Travis`, hasta hace poco tenía buen soporte, pero ahora han cambiado de equipo y está algo peor (pj en `go` no tenían la última versión, pero creo que para `node` sí)

Muchas gracias por tu review @danieeeld2!!! 😄

migueruiz commented:
@JJ Listo para revisión!!

JJ reviewed (changes requested):
Usar una versión fija no acaba de ser buena idea; en todo caso también conviene que contemples otras posibilidades en sistemas de integración continua.

migueruiz committed:
Modifico el fichero /doc/eleccion_ci.md. closes #44.

Modifico error Typo

migueruiz committed:
Modifico el fichero /doc/eleccion_ci.md. closes #44.

Añado otra opción más para tener mas elecciones.

migueruiz committed:
Modifico el fichero /doc/eleccion_ci.md. closes #44.

Modifico la decisión en función de la nueva opción que he encontrado.

migueruiz committed:
Modifico el fichero node.yml para GitHub Actions. closes #46.

Cambio la versión de node que se comprueba, poniendo varias versiones, con el uso de una matrix.

migueruiz commented:
@JJ Listo para revisión!!

migueruiz committed:
Modifico el fichero /doc/eleccion_ci.md. closes #44.

Modifico el fichero en función de la nueva opción que he encontrado.

danieeeld2 commented on doc/eleccion_ci.md:
```diff
@@ -28,6 +28,10 @@ Es una buena opción como la anterior, tiene una fácil integración con Github
 
 Es compatible con Github. Además nos permite un entorno de ejecucción de imagenes docker contando también con un soporte con problemas o errores en aquellas que sean remotas. Otra de sus ventajas es el ciclo de actualizaciones, ya que encontramos que su ultima versión data de julio de 2023.
 
+* [Buddy](https://buddy.works/)
```
😲 no había visto este antes

migueruiz committed:
Modifico el fichero node.yml para GitHub Actions. closes #46.

Cambio la versión de node poniendo que use las versiones lts, y la actual de ese momento.

JJ reviewed (approved):
👍

migueruiz committed:
Modifico el fichero node.yml para GitHub Actions. closes #46.

Cambio la versión de node poniendo que use las versiones lts, y la actual de ese momento.

migueruiz committed:
Modifico el fichero node.yml para GitHub Actions. closes #46.

Cambio la versión de node poniendo que use las versiones lts, y la actual de ese momento.

migueruiz committed:
Modifico el fichero node.yml para GitHub Actions. closes #46.

Cambio la versión de node poniendo que use las versiones lts, y la actual de ese momento.

migueruiz committed:
Modifico el fichero node.yml para GitHub Actions. closes #46.

Cambio la versión de node poniendo que use las versiones lts, la anterior lts, y la actual de ese momento.
